### PR TITLE
IBX-5948: 'Create Custom Name Schema strategy' page added in PIM

### PR DESCRIPTION
**Pull request by julitafalcondusza**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | [IBX-5948](https://issues.ibexa.co/browse/IBX-5948)
| Versions      | master
| Edition       | Content

'Create Custom Name Schema strategy' page added in PIM

#### Checklist

- [x] Text renders correctly
- [x] Text has been checked with vale
- [x] Description metadata is up to date
- [x] Code samples are working
- [x] PHP code samples have been fixed with PHP CS fixer
- [ ] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: code_samples/catalog/custom_name_schema_strategy/NameSchema/NameSchemaStrategyInterface.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?php declare(strict_types=1);
```
Comment (adamwojs) [inline code_samples/catalog/custom_name_schema_strategy/NameSchema/NameSchemaStrategyInterface.php:1]:
```suggestion
<?php 

declare(strict_types=1);
```

File: code_samples/catalog/custom_name_schema_strategy/NameSchema/NameSchemaStrategyInterface.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace App\Attribute\NameSchema;
+
+use Ibexa\Contracts\ProductCatalog\Values\AttributeDefinitionInterface;
+use Ibexa\ProductCatalog\NameSchema\NameSchemaStrategyInterface;
+
+final class NameSchemaStrategyInterface implements NameSchemaStrategyInterface
```
Comment (adamwojs) [inline code_samples/catalog/custom_name_schema_strategy/NameSchema/NameSchemaStrategyInterface.php:8]:
```suggestion
final class CustomNameSchemaStrategy implements NameSchemaStrategyInterface
```

File: code_samples/catalog/custom_name_schema_strategy/src/bundle/Resources/config/services/name_schema.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+services:
+    _defaults:
+        public: false
+        autowire: true
+        autoconfigure: true
+
+    App\Attribute\NameSchema\CustomNameSchemaStrategy: ~
```
Comment (adamwojs) [inline code_samples/catalog/custom_name_schema_strategy/src/bundle/Resources/config/services/name_schema.yaml:7]:
```suggestion
    App\Attribute\NameSchema\CustomNameSchemaStrategy: ~
    
```

File: docs/pim/create_custom_name_schema_strategy.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+description: Create CustomNameSchemaStrategy to generate URL aliases based on attribute values.
+---
+
+# Create custom Name Schema strategy
+
+Make sure you have properly configured attributes. Each attribute that you want to include in the URL alias must have a name schema strategy.
+
+## Create converting class
+
+Start by creating a `CustomNameSchemaStrategy` class, which implements `\Ibexa\ProductCatalog\NameSchema\NameSchemaStrategyInterface`. This class is responsible for converting attribute values into a string of URL parameters:
```
Comment (adamwojs) [inline docs/pim/create_custom_name_schema_strategy.md:11]: If `\Ibexa\ProductCatalog\NameSchema\NameSchemaStrategyInterface` is extension point (it should be) then it needs to be moved to contracts // cc @kisztof

File: mkdocs.yml [hunk 5]
```diff
@@ -306,6 +306,7 @@ nav:
         - Create custom attribute type: pim/create_custom_attribute_type.md
         - Create product code generator: pim/create_product_code_generator.md
         - Create custom catalog filter: pim/create_custom_catalog_filter.md
+        - Create Custom Name Schema strategy: pim/create_custom_name_schema_strategy.md
```
Comment (adamwojs) [inline mkdocs.yml:309]: Capitalization is inconsistent

File: code_samples/catalog/percent_name_schema_strategy/NameSchema/NameSchemaStrategyInterface.php [hunk 6]
```diff
@@ -0,0 +1,19 @@
+<?php declare(strict_types=1);
+
+namespace App\Attribute\Percent;
+
+use Ibexa\Contracts\ProductCatalog\NameSchema\NameSchemaStrategyInterface;
+use Ibexa\Contracts\ProductCatalog\Values\AttributeDefinitionInterface;
+
+final class NameSchemaStrategyInterface implements NameSchemaStrategyInterface
```
Comment (adamwojs) [inline code_samples/catalog/percent_name_schema_strategy/NameSchema/NameSchemaStrategyInterface.php:8]:
```suggestion
final class PercentNameSchemaStrategy implements NameSchemaStrategyInterface
```

**Review by adamwojs — APPROVED** [review]

File: docs/pim/create_custom_name_schema_strategy.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
+---
+description: Create custom Name Schema strategy to generate URL aliases based on attribute values.
+---
+
+# Create custom Name Schema strategy
+
+Make sure you the attributes are configured correctly. Each attribute that you want to include in the URL alias must have a name schema strategy.
```
Comment (MagdalenaZuba) [inline docs/pim/create_custom_name_schema_strategy.md:7]:
```suggestion
Make sure the attributes are configured correctly.
Each attribute that you want to include in the URL alias must have a name schema strategy.
```

File: docs/pim/create_custom_name_schema_strategy.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
+---
+description: Create custom Name Schema strategy to generate URL aliases based on attribute values.
+---
+
+# Create custom Name Schema strategy
```
Comment (MagdalenaZuba) [inline docs/pim/create_custom_name_schema_strategy.md:5]:
```suggestion
# Create custom name schema strategy
```

File: docs/pim/create_custom_name_schema_strategy.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
+---
+description: Create custom Name Schema strategy to generate URL aliases based on attribute values.
```
Comment (MagdalenaZuba) [inline docs/pim/create_custom_name_schema_strategy.md:2]:
```suggestion
description: Create custom name schema strategy to generate URL aliases based on attribute values.
```

File: docs/pim/create_custom_name_schema_strategy.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+## Register strategy
+
+Next, you need to register the strategy in the dependency injection container:
+
+``` yaml
+[[= include_file('code_samples/catalog/percent_name_schema_strategy/src/bundle/Resources/config/services/name_schema.yaml') =]]
+```
+
+This ensures that the custom Name Schema strategy is available for use in generating URL aliases based on attribute values.
```
Comment (MagdalenaZuba) [inline docs/pim/create_custom_name_schema_strategy.md:26]:
```suggestion
This ensures that the custom name schema strategy is available for use in generating URL aliases based on attribute values.
```

**Review by MagdalenaZuba — CHANGES_REQUESTED** [review]

File: docs/pim/create_custom_name_schema_strategy.md [hunk 11]
```diff
@@ -0,0 +1,27 @@
+---
+description: Create custom name schema strategy to generate URL aliases based on attribute values.
+---
+
+# Create custom name schema strategy
+
+Make sure the attributes are configured correctly.
```
Comment (dabrt) [inline docs/pim/create_custom_name_schema_strategy.md:7]: Why are we doing it? What problem are we solving or what improvement implementing?

File: mkdocs.yml [hunk 12]
```diff
@@ -306,6 +306,7 @@ nav:
         - Create custom attribute type: pim/create_custom_attribute_type.md
         - Create product code generator: pim/create_product_code_generator.md
         - Create custom catalog filter: pim/create_custom_catalog_filter.md
+        - Create custom Name Schema strategy: pim/create_custom_name_schema_strategy.md
```
Comment (MagdalenaZuba) [inline mkdocs.yml:309]:
```suggestion
        - Create custom name schema: pim/create_custom_name_schema_strategy.md
```

**Review by MagdalenaZuba — APPROVED** [review]

**Review by dabrt — APPROVED** [review]

**Pull request merged**